http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpJupyterDAO.scala

```diff
@@ -48,9 +50,43 @@ class HttpJupyterDAO[F[_]: Timer: ContextShift: Concurrent](val runtimeDnsCache:
               )
             )
           } yield res.forall(k => k.kernel.executionState == Idle)
-        case _ => Concurrent[F].pure(true)
+        case _ => F.pure(true)
       }
     } yield resp
+
+  override def createTerminal(googleProject: GoogleProject, runtimeName: RuntimeName): F[Unit] =
+    Proxy.getRuntimeTargetHost[F](runtimeDnsCache, googleProject, runtimeName) flatMap {
+      case HostReady(targetHost) =>
+        client
+          .successful(
+            Request[F](
+              method = Method.POST,
+              uri = Uri.unsafeFromString(
+                s"https://${targetHost.toString}/notebooks/${googleProject.value}/${runtimeName.asString}/api/terminals"
+              )
+            )
+          )
+          .flatMap(res => if (res) F.unit else logger.error("Fail to create new terminal"))
+      case _ => F.unit
+    }
+
+  override def ifTerminalExist(googleProject: GoogleProject,
```

Comment:
I wonder if 
```suggestion
  override def terminalExists(googleProject: GoogleProject,
```
would read more naturally?

Reply:
I tend to add `is` or `if` if the return type is a `Boolean`...I can add `s` at the end to make it reads better

Reply:
I guess this may be a personal convention. I also take into account the call site, which, in the case of methods returning `Boolean` tends to have an `if` clause so the method name containing another `if` doesn't read well...but I don't feel strongly 🤷‍♂️

Reply:
renamed